evm/hardhat.config.ts

```diff
@@ -31,27 +44,58 @@ task("set-metadata-ft", "Set metadata for NEP-141 tokens on the Ethereum side")
 
 task("deploy-bridge-token-factory", "Deploys the OmniBridge contract")
   .addParam("bridgeTokenImpl", "The address of the bridge token implementation")
-  .addParam("nearBridgeDerivedAddress", "The derived EVM address of the Near's OmniBridge")
-  .addParam("omniBridgeChainId", "Chain Id of the network in the OmniBridge")
+  .addParam("nearBridgeAccountId", "The OmniBridge account ID on NEAR")
   .setAction(async (taskArgs, hre) => {
     const { ethers, upgrades } = hre
+    const networkConfig = hre.network.config as HttpNetworkUserConfig
+    const omniChainId = networkConfig.omniChainId
+    const wormholeAddress = networkConfig.wormholeAddress
+
+    const mpcRootPublicKey = hre.network.name.endsWith("mainnet")
+      ? mpcRootPublicKeys.mainnet.key
+      : mpcRootPublicKeys.testnet.key
+
+    const nearBridgeDerivedAddress = await deriveEthereumAddress(
+      taskArgs.nearBridgeAccountId,
+      "bridge-1",
+      mpcRootPublicKey,
+    )
+
+    console.log(`Derived addres: ${nearBridgeDerivedAddress}`)
+    console.log(`Omni chain id: ${omniChainId}`)
+
+    const isWormholeContract = !hre.network.name.startsWith("eth")
+    const contractName = isWormholeContract ? "OmniBridgeWormhole" : "OmniBridge"
+    const OmniBridgeContract = await ethers.getContractFactory(contractName)
+    const consistencyLevel = 0
 
-    const OmniBridgeContract = await ethers.getContractFactory("OmniBridge")
     const OmniBridge = await upgrades.deployProxy(
       OmniBridgeContract,
-      [taskArgs.bridgeTokenImpl, taskArgs.nearBridgeDerivedAddress, taskArgs.omniBridgeChainId],
+      isWormholeContract
+        ? [
+            taskArgs.bridgeTokenImpl,
+            nearBridgeDerivedAddress,
+            omniChainId,
+            wormholeAddress,
+            consistencyLevel,
+          ]
+        : [taskArgs.bridgeTokenImpl, nearBridgeDerivedAddress, omniChainId],
       {
-        initializer: "initialize",
+        initializer: isWormholeContract ? "initializeWormhole" : "initialize",
         timeout: 0,
       },
     )
 
     await OmniBridge.waitForDeployment()
-    console.log(`OmniBridge deployed at ${await OmniBridge.getAddress()}`)
+    const bridgeAddress = await OmniBridge.getAddress()
+    const wormholeAddressStorageValue = await hre.ethers.provider.getStorage(bridgeAddress, 58)
+
+    console.log(`OmniBridge deployed at ${bridgeAddress}`)
     console.log(
       "Implementation address:",
       await upgrades.erc1967.getImplementationAddress(await OmniBridge.getAddress()),
     )
+    console.log(`Wormhole address ${wormholeAddressStorageValue}`)
```

Comment:
Should this match the `wormholeAddress` from the config?
Should we add an additional check to ensure it’s as expected?

Reply:
1df4370a93d78233196a8178bc764d5b13687aae